JEasyCryptoClient/src/CryptoClient.java

```diff
@@ -137,39 +140,38 @@ private String enterText(String prompt, boolean required) {
 	}
 	
 	private void handleCapabilityRequest() throws IOException {
-		JSONObject request = new JSONObject();
-		request.put("id", requestId++);
-		request.put("operation", "capabilities");
-		String data = request.toJSONString();
+		String data = createRequest("capabilities", null, null);
 		DatagramPacket packet = new DatagramPacket(data.getBytes(), data.length(), serverAddr, serverPort);
 		socket.send(packet);
 	}
 	
 	private void handleEncryptRequest() throws IOException {
-		JSONObject request = new JSONObject();
-		request.put("id", requestId++);
-		request.put("operation", "encrypt");
 		String method = enterText("Give encryption method", true);
 		String text = enterText("Give text to encrypt", false);
-		request.put("method", method);
-		request.put("data", text);
-		String data = request.toJSONString();
+		String data = createRequest("encrypt", method, text);
 		DatagramPacket packet = new DatagramPacket(data.getBytes(), data.length(), serverAddr, serverPort);
 		socket.send(packet);
 	}
 	
 	private void handleDecryptRequest() throws IOException {
-		JSONObject request = new JSONObject();
-		request.put("id", requestId++);
-		request.put("operation", "decrypt");
 		String method = enterText("Give decryption method", true);
 		String text = enterText("Give text to decrypt", false);
-		request.put("method", method);
-		request.put("data", text);
-		String data = request.toJSONString();
+		String data = createRequest("decrypt", method, text);
 		DatagramPacket packet = new DatagramPacket(data.getBytes(), data.length(), serverAddr, serverPort);
 		socket.send(packet);
 	}
+
+	private String createRequest(String operation, String method, String text) {
+		Map<String, Object> requestMap = new HashMap<>();
```

Comment:
I think it would be better to use an `HashMap` object directly instead of `Map`. I think we can be more specific at this point,